src/usr/local/buildpack/tools/yarn.sh

```diff
@@ -7,11 +7,7 @@ check_command node
 tool_path=$(find_versioned_tool_path)
 
 if [[ -z "${tool_path}" ]]; then
-  INSTALL_DIR=$(get_install_dir)
-  base_path=${INSTALL_DIR}/${TOOL_NAME}
-  tool_path=${base_path}/${TOOL_VERSION}
-
-  mkdir -p "${tool_path}"
+  tool_path="$(create_versioned_tool_path)"
 
   NPM_CONFIG_PREFIX=$tool_path npm install --cache /tmp/empty-cache -g "${TOOL_NAME}@${TOOL_VERSION}"
```

Comment:
We need to clean cache. Looks like it's not removed from install-tool, so next install command with different user fails.

Maybe better use a random path?

Reply:
Seems we need to clean this path `/home/user/.npm/_cacache`

Reply:
Seems we need to do a `npm cache clean --force` previously to the install

Reply:
I thought when using the cache param it will use only that path, but npm seems to still use home dir.

Reply:
Looks like a npm bug

- https://github.com/npm/cli/issues/3350